### ISSUE TITLE
Database pool connections not reused within a single distributed transaction

### ISSUE THREAD

**Issue by TormundTargers** [atomikos/transactions-essentials#103]
**Describe the bug**
I am using transactions-remoting with the JAXRS filters to implement distributed transactions.

It appears that a single database connection is used from the pool when making multiple connections to the database when they are done within a transaction in a single system.

However, if a transaction is started in remote system A, then system A makes 3 HTTP requests to system B. Then system B makes 3 database connections and does not release them to the pool until the transaction in system A is committed.

A new connection from the database pool is created for every database interaction within a single remote transaction.

**To Reproduce**
1. Start transaction in system A.
2. Make 3 http requests from system A to system B
 Where each request causes an interaction with a database.
3. Commit transaction in system A.
4. Observe that system B keeps 3 connections to the database open until the transaction is committed.

I expected that only one database connection would be used. Because that is what happens if you start a transaction on system B, make 3 database interactions, and commit the transaction.

 **Extreme transactions version**
5.0.82

**Additional context**
I am using an Oracle 11g database.

I can link to a small example project that can demonstrate the issue.

**Comment by GuyPardon** [comment]
Hi,

Thanks - an example project would be nice, and also the atomikos logs (in DEBUG mode) of your scenario. That way, we can check what is going on.

Guy

**Comment by TormundTargers** [comment]
You will obviously need to create the tables and users yourself to be able to execute but here is a project and some logs of either case.

## Project
[atomikos-test-example.tar.gz](https://github.com/atomikos/transactions-essentials/files/3987975/atomikos-test-example.tar.gz)

### Distributed example
[client.log](https://github.com/atomikos/transactions-essentials/files/3987978/client.log)
[server.log](https://github.com/atomikos/transactions-essentials/files/3987980/server.log)

**Steps**
1. Run `server.Main`
2. Run `client.DistributedTransactionClient`
3. Observe that 4 connections are made to the database, which are eventually released after 20 seconds due to the data source's max idle time. I am checking this via running:
```sql
SELECT STATE, STATUS, PREV_EXEC_START, WAIT_CLASS
FROM V$SESSION
WHERE USERNAME = 'ATOMIKOS_TEST';
```

### Non-distributed example
[client.log](https://github.com/atomikos/transactions-essentials/files/3987989/client.log)
[server.log](https://github.com/atomikos/transactions-essentials/files/3987990/server.log)

**Steps**
1. Run `server.Main`
2. Run `client.NonDistributedTransactionClient`
3. Observe that only 1 connection is made to the database. This was checked using the query above.

**Comment by GuyPardon** [comment]
Fixed in 5.0.83
